MSBuild.Caching/CacheSet.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace MSBuild.Caching;
+
+public class CacheSet : Task
+{
+    public string? Key { get; set; }
+
+    public string? Value { get; set; }
+
+    public override bool Execute()
+    {
+        this.BuildEngine4.RegisterTaskObject(
+            this.Key, this.Value, RegisteredTaskObjectLifetime.Build, allowEarlyCollection: false);

```

Comment:
Since it's only a version string, I thought to switch off early collection, as the cost of running MinVer again seems more than keeping the string around longer.

From https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework.ibuildengine4.registertaskobject (emphasis mine) —

> It is strongly recommended that allowEarlyCollection be set to true _if the object will retain any significant amount of data…_

Reply:
MSBuild doesn't actually pay attention to this any more:

https://github.com/dotnet/msbuild/blob/8b03742aafd18a26a5d03cd045e4d24963886503/src/Shared/RegisteredTaskObjectCacheBase.cs#L59-L64

(We should update the docs though: https://github.com/dotnet/msbuild/issues/10425

Reply:
Ah, thanks for the info!